ParkingLot/com/parkinglot/ParkingLotService.java

```diff
@@ -0,0 +1,147 @@
+package com.parkinglot;
+
+import com.parkinglot.models.Floor;
+import com.parkinglot.models.ParkingLot;
+import com.parkinglot.models.Slot;
+import com.parkinglot.models.Vehicle;
+import com.parkinglot.models.VehicleType;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ParkingLotService {
+  private Map<Floor, List<Slot>> floorSlotMap = new LinkedHashMap<>();
+  private ParkingLot parkingLot = null;
+  public void createParkingLot(String parkingId,  int numberOfFloors, int numberOfSlots ) {
+    parkingLot = new ParkingLot(parkingId);
+    for (int floorNumber = 0; floorNumber < numberOfFloors; floorNumber++) {
+      List<Slot> slots = new ArrayList<>();
+      for (int slotNumber = 0; slotNumber < numberOfSlots; slotNumber++) {
+        VehicleType vehicleType;
+        Slot slot;
+        if (slotNumber == 0) {
+          vehicleType = VehicleType.TRUCK;
+          slot = new Slot(slotNumber, vehicleType);
+        } else if (slotNumber == 1 || slotNumber == 2) {
+          vehicleType = VehicleType.BIKE;
+          slot = new Slot(slotNumber, vehicleType);
+        } else {
+          vehicleType = VehicleType.CAR;
+          slot = new Slot(slotNumber, vehicleType);
+        }
+        slots.add(slot);
+      }
+      Floor floor = new Floor(floorNumber);
+      floorSlotMap.put(floor,slots);
+    }
+    System.out.println("Created a parking lot with " + numberOfFloors + " floors and " + numberOfSlots + " per floor");
+
+  }
+
+  public void displaySlots(String displayType, String vehicleTypeToCheck) {
+    switch (displayType) {
+      case "free_count" -> displayAvailableSlotsCount(vehicleTypeToCheck);
```

Comment:
Magic strings?

This should be handled through enums if you want to do like this. Ideally, it should have been done through different public methods being exposed from here instead of taking the type as parameter.

Reply:
@gcnit : I am not totally clear how to follow this step : “Ideally, it should have been done through different public methods being exposed from here instead of taking the type as parameter.”

Reply:
displayslots method should not exist. The functions that you're calling in each case should be called directly by the caller.